queries/nix.scm

```diff
@@ -0,0 +1,146 @@
+;; Statements
+
+[
+    (binding)
+] @statement
+;; This matches assert statements as stand-alone statements. This
+;; is because it's common to have a series of assert statements in a file, which
+;; would otherwise all match as a single statement. See the playground/nix/asserts.nix
+;; file for an example.
+(assert_expression
+    "assert" @statement.start
+    condition: (_)
+    ";" @statement.end
+    body: (_)
+)
+(binding_set) @map.iteration @list.iteration @ifStatement.iteration
+(source_code) @statement.iteration @map.iteration @comment.iteration
+
+;;!! let
+;;!!   a = 1;
+;;!    <*****
+;;!!   b = 2;
+;;!    *****>
+;;!! in a + b
+(let_expression
+    "let"
+    (_) @statement.iteration @value.iteration @name.iteration
+)
+
+;; Conditionals
+
+;;!! key = if a then b else c;
+;;!        ^^^^^^^^^^^^^^^^^^
+(if_expression) @ifStatement @branch.iteration
+;;!! key = if a then b else c;
+;;!        ^^^^
+(if_expression
+    "if" @branch.start
+    condition: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!             ^^^^^^
+(if_expression
+    "then" @branch.start
+    consequence: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!                    ^^^^^^
+(if_expression
+    "else" @branch.start
+    alternative: (_) @branch.end
+)
+
+;; Lists and maps
+
+;;!! foo = [ a b c ];
+;;!        ^^^^^^^^^
+(list_expression) @list
+
+;;!! foo = { x = 1; y = 2; };
+;;!        ^^^^^^^^^^^^^^^^^^
+[
+    (attrset_expression)
+    (rec_attrset_expression)
+] @map @statement.iteration @value.iteration @name.iteration
+
+;;!! foo = { x = 1; y = 2; };
+;;!          ^
```

Comment:
test please

Reply:
Done. Also I tweaked this to add a `@_.domain` because accessing the key wouldn't work in the RHS which I think is wrong, so added a test for that as well.